src/core/thread/topology.cpp

```diff
@@ -386,7 +386,7 @@ bool Child::HasIp6Address(const Ip6::Address &aAddress) const
     return retval;
 }
 
-#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+#if (OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE)
```

Comment:
Keep consistent with other conditions.
```suggestion
#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
```